service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/exceptions/IotHubExceptionManager.java

```diff
@@ -130,4 +130,64 @@ else if (responseStatus > 300)
         }
         // Codes_SRS_SERVICE_SDK_JAVA_IOTHUBEXCEPTIONMANAGER_12_012: [The function shall return without exception if the response status equal or less than 300]
     }
+
+    /**
+     * Return a new exception instance that best matches the given HTTP status code and description
+     * @param responseStatus The HTTP status code (404, 500, etc.)
+     * @param description The HTTP response body
+     * @return a new exception instance that best matches the given HTTP status code
+     */
+    public static IotHubException mapException(int responseStatus, String description)
```

Comment:
I needed to add this function because the only other error code to exception mapping logic takes in an httpResponse as the argument. For the AMQP CBS code path, there is no httpResponse object to give it